app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,101 @@
+from django.test import TestCase
+from rest_framework import status
+
+from rest_framework.test import APIClient
+
+from recipe.models import Recipe, Ingredient
+from recipe.serializers import RecipeSerializer
+import json
+
+
+def sample_recipe():
+    """Create and return a sample recipe"""
+    defaults = {
+        "name": "Sample recipe",
+        "description": "instructions for the recipe",
+    }
+    recipe = Recipe.objects.create(**defaults)
+    recipe.ingredients.create(name="ingredient1")
+    recipe.ingredients.create(name="ingredient2")
+
+    return recipe
+
+
+class RecipeApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_retrieve_recipes(self):
+        sample_recipe()
+        res = self.client.get("/recipes/")
+
+        recipes = Recipe.objects.all().order_by("-id")
+        serializer = RecipeSerializer(recipes, many=True)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_retrieve_recipe(self):
+        recipe = sample_recipe()
+        res = self.client.get(f"/recipes/{recipe.id}/")
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data["name"], recipe.name)
+
+    def test_retrieve_filtered_recipes(self):
+        sample_recipe()
+        res = self.client.get("/recipes/?name=mple")
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(res.data), 1)
```

Comment:
You only create one recipe in the test and you test that you retrieve one. It would be more robust to create a couple of recipes and test that you get only the matching one.

Reply:
Done